sql/src/main/java/io/crate/jobs/PageBucketReceiver.java

```diff
@@ -56,6 +56,14 @@
  */
 public interface PageBucketReceiver {
 
+    /**
+     * Receives a bucket from an upstream which holds result data. This method may
+     * be called multiple times with the same bucketIdx if there are multiple pages.
+     * @param bucketIdx A bucket id which uniquely identifies all buckets of this page.
+     * @param rows The bucket which holds result rows.
+     * @param isLast Indicates whether this is the last bucket with this id.
+     * @param pageResultListener The ResultListener which is informed if more data is available.
```

Comment:
this is missleading, it's not if more data is available, it's that more data is required. (As stated in https://github.com/crate/crate/pull/6681/files#diff-173f7d1183bba8fca495fad1939d503fR53)

Reply:
It's both at the same time because the context knows that it hasn't received the `isLast` flag yet. Shouldn't go here in the interface though. I'll rephrase to make it clear.